kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/WatchConnectionManager.java

```diff
@@ -289,8 +289,8 @@ public void execute() {
                 // An unexpected error occurred and we didn't even get an onFailure callback.
                 logger.error("Exception in reconnect", e);
                 webSocketRef.set(null);
+                closeEvent(new KubernetesClientException("Unhandled exception in reconnect attempt", e));
```

Comment:
The reordering was because we allow `onClose()` to be called exactly once. The fix is `closeEvent(null);` within the `close()` method, which ensures that in case of a user initiated closure, we do try and call `onClose()` if we haven't previously.